docs/data/data-grid/column-menu/column-menu.md

```diff
@@ -0,0 +1,138 @@
+### Add, override, hide and reorder items
+
+Column menu component accept `components`, `componentsProps` and `initialItems` using which you can:
+
+- Add new items
+- Override default items
+- Hide default items
+- Reorder items
+
+**components**: Prop accepted by column menu components `<GridColumnMenuDefault />` or `<GridColumnMenuSimple />`, could be used to override default column menu components or add new components.
+
+Default components supported by community package are `ColumnMenuSortItem`, `ColumnMenuFilterItem`, `ColumnMenuHideItem` and `ColumnMenuColumnsItem`. Pro package adds `ColumnMenuPinningItem` and Premium package adds `ColumnMenuAggregationItem` and `ColumnMenuGroupingItem`.
+
+**componentsProps**: Every item has a `displayOrder` based which it will be placed before or after other items in the menu. It can be overriden using `componentsProps`. You can also use this to pass extra props to column menu components.
+
+`componentsProps` for a component uses the same key but in _camelCase_. E.g for `ColumnMenuFilterItem`, the componentsProps key will be `columnMenuFilterItem`.
+
+**initialItems**: Visibility of default items is controlled using respective feature hooks, for cutom items though, use `initialItems` to show the items added by passing to `components`. All the new items will be placed in the end (default `displayOrder: 100`), unless customized by `displayOrder`.
+
+```tsx
+function CustomColumnMenu(props: GridColumnMenuProps) {
+  return (
+    <GridColumnMenuDefault
+      {...props}
+      components={{
+        // Override slot for `ColumnMenuFilterItem`
+        ColumnMenuFilterItem: CustomFilterItem,
+        // Hide `ColumnMenuColumnsItem`
+        ColumnMenuColumnsItem: null,
+        // Add new item
+        ColumnMenuUserItem: CustomUserItem,
+      }}
+      componentsProps={{
+        // Swap positions of filter and sort items
+        columnMenuFilterItem: {
+          displayOrder: 0, // Previously `10`
+        },
+        columnMenuSortItem: {
+          displayOrder: 10, // Previously `0`
+        },
+        columnMenuUserItem: {
+          // set `displayOrder` for new item
+          displayOrder: 15,
+          // pass additional props
+          myCustomValue: 'Do custom action',
+          myCustomHandler: () => alert('Custom handler fired'),
+        },
+      }}
+      initialItems={['ColumnMenuUserItem']}
+    />
+  );
+}
+```
+
+{{"demo": "ReuseColumnMenuGrid.js", "bg": "inline"}}
+
+#### Display orders for default items
+
+As a reference, here is the order for each of the default items, there are some gaps to be able to place items in between:
+
+| **Component**             | **Package** | **Display order (default design)** | **Display order (simple design)** |
+| ------------------------- | ----------- | ---------------------------------: | --------------------------------: |
+| ColumnMenuSortItem        | Community   |                                  0 |                                 0 |
```

Comment:
Maybe combine the two tables on this page into one Data Grid, with menu item components as rows, grouped by column menu type. We can mark Pro/Premium item with the appropriate icon - similar to http://localhost:3001/x/react-data-grid/selection/#apiref

Reply:
That seems a good idea, why not use our own Grid when we have a better alternative to markdown table 🎉

Reply:
Maybe use plan icons with links instead of the Plan column?
<img width="835" alt="Screenshot 2022-11-30 at 15 03 30" src="https://user-images.githubusercontent.com/13808724/204816248-ebac8270-e39f-438d-a1f9-3d6055ee9edf.png">

I've also added some styles to the component `renderCell` to make it more similar to the `code` block in the markdown.

Here's a git patch if you agree with the changes:
```diff
diff --git a/docs/data/data-grid/column-menu/ColumnMenuGridReferences.tsx b/docs/data/data-grid/column-menu/ColumnMenuGridReferences.tsx
index 3454d7c..e7963bb 100644
--- a/docs/data/data-grid/column-menu/ColumnMenuGridReferences.tsx
+++ b/docs/data/data-grid/column-menu/ColumnMenuGridReferences.tsx
@@ -5,42 +5,44 @@ import {
   useKeepGroupedColumnsHidden,
   GridRenderCellParams,
 } from '@mui/x-data-grid-premium';
-import Chip from '@mui/material/Chip';
-import Avatar from '@mui/material/Avatar';
 import Stack from '@mui/material/Stack';
 import Typography from '@mui/material/Typography';
 
-const getChipProperties = (plan: string) => {
+const getPlanProps = (plan: string) => {
   switch (plan.toLowerCase()) {
     case 'premium':
-      return { avatarLink: '/static/x/premium.svg', color: '#ffecc8' };
+      return {
+        href: '/x/introduction/licensing/#premium-plan',
+        className: 'plan-premium',
+        title: 'Premium plan',
+      };
     case 'pro':
-      return { avatarLink: '/static/x/pro.svg', color: '#c8e9ff' };
+      return {
+        href: '/x/introduction/licensing/#pro-plan',
+        className: 'plan-pro',
+        title: 'Pro plan',
+      };
     default:
-      return { avatarLink: undefined, color: '#c8ffdb' };
+      return null;
   }
 };
 
-function PlanTag(props: { plan?: string }) {
+function PlanIcon(props: { plan?: string }) {
   if (!props.plan) {
     return null;
   }
-  const chipPropperties = getChipProperties(props.plan);
-  const avatar = !chipPropperties.avatarLink ? undefined : (
-    <Avatar src={chipPropperties.avatarLink} />
-  );
+  const planProps = getPlanProps(props.plan);
+  if (!planProps) {
+    return null;
+  }
   return (
-    <Stack sx={{ width: '100%' }}>
-      <Chip
-        avatar={avatar}
-        sx={{ background: chipPropperties.color, color: 'rgba(0, 0, 0, 0.87)' }}
-        label={props.plan}
-      />
-    </Stack>
+    <a href={planProps.href} target="_blank" rel="noreferrer">
+      <span className={planProps.className} title={planProps.title} />
+    </a>
   );
 }
 
-function ComponentTag(props: { value?: string }) {
+function ComponentTag(props: { value?: string; plan?: string }) {
   if (!props.value) {
     return null;
   }
@@ -48,16 +50,22 @@ function ComponentTag(props: { value?: string }) {
   return (
     <Stack gap={0.5}>
       {components.map((c, key) => (
-        <Typography
-          key={key}
-          sx={{
-            borderRadius: '5px',
-            background: 'rgba(102, 178, 255, 0.15)',
-            fontSize: '0.8rem',
-          }}
-        >
-          {c}
-        </Typography>
+        <div>
+          <Typography
+            key={key}
+            sx={{
+              borderRadius: '5px',
+              background: 'rgba(102, 178, 255, 0.15)',
+              fontSize: '0.8rem',
+              fontFamily: 'Consolas, "Liberation Mono", Menlo, Courier, monospace',
+              padding: '0 5px',
+              display: 'inline-block',
+            }}
+          >
+            {c}
+          </Typography>
+          <PlanIcon plan={props.plan} />
+        </div>
       ))}
     </Stack>
   );
@@ -67,20 +75,10 @@ const columns = [
   {
     field: 'slot',
     headerName: 'Component',
-    width: 220,
     renderCell: (params: GridRenderCellParams<string>) => (
-      <ComponentTag value={params.value} />
-    ),
-  },
-  {
-    field: 'plan',
-    headerName: 'Plan',
-    width: 170,
-    type: 'singleSelect',
-    valueOptions: ['Premium', 'Pro', 'Community'],
-    renderCell: (params: GridRenderCellParams<string>) => (
-      <PlanTag plan={params.value} />
+      <ComponentTag value={params.value} plan={params.row.plan} />
     ),
+    width: 240,
   },
   { field: 'design', headerName: 'Menu Design' },
   {
@@ -217,11 +215,6 @@ export default function ColumnMenuGridReferences() {
   const initialState = useKeepGroupedColumnsHidden({
     apiRef,
     initialState: {
-      columns: {
-        columnVisibilityModel: {
-          plan: false,
-        },
-      },
       rowGrouping: {
         model: ['design'],
       },
@@ -240,10 +233,8 @@ export default function ColumnMenuGridReferences() {
         hideFooter
         isGroupExpandedByDefault={(node) => node.groupingKey === 'Default'}
         groupingColDef={{
-          headerName: 'Design & Plan',
+          headerName: 'Component slot',
           hideDescendantCount: true,
-          leafField: 'plan',
-          width: 140,
         }}
       />
     </div>

```